feat: add C `ndarray` API and code refactor to `blas/ext/base/dapxsumpw` (#2937)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/ext/base/dapxsumpw/README.md

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/blas/ext/base/dapxsumpw.h"
```

#### stdlib_strided_dapxsumpw( N, alpha, \*X, strideX )

> Add a scalar constant to each double-precision floating-point strided array element and compute the sum using pairwise summation.

Reviewer: ```suggestion
Adds a scalar constant to each double-precision floating-point strided array element and computes the sum using pairwise summation.
```

```c
const double x[] = { 1.0, 2.0, 3.0, 4.0 };

> double v = stdlib_strided_dapxsumpw( 4, 5.0, x, 1 );

Reviewer: ```suggestion
double v = stdlib_strided_dapxsumpw( 4, 5.0, x, 1 );
// returns 30.0
```

```

The function accepts the following arguments:

-   **N**: `[in] CBLAS_INT` number of indexed elements.
-   **alpha**: `[in] double` scalar constant.
-   **X**: `[in] double*` input array.
-   **strideX**: `[in] CBLAS_INT` index increment for `X`.

```c
double stdlib_strided_dapxsumpw( const CBLAS_INT N, const double alpha, const double *X, const CBLAS_INT strideX );
```

#### stdlib_strided_dapxsumpw_ndarray( N, alpha, \*X, strideX, offsetX )

> Add a scalar constant to each double-precision floating-point strided array element and compute the sum using pairwise summation and alternative indexing semantics.

Reviewer: ```suggestion
Adds a scalar constant to each double-precision floating-point strided array element and computes the sum using pairwise summation and alternative indexing semantics.
```

```c
const double x[] = { 1.0, 2.0, 3.0, 4.0 };

> double v = stdlib_strided_dapxsumpw_ndarray( 4, 5.0, x, 1, 0 );

Reviewer: ```suggestion
double v = stdlib_strided_dapxsumpw_ndarray( 4, 5.0, x, 1, 0 );
// returns 30.0
```

const double x[] = { 1.0, 2.0, 3.0, 4.0 };

double v = stdlib_strided_dapxsumpw_ndarray( 4, 5.0, x, 1, 0 );
// returns 30.0
```

The function accepts the following arguments:

-   **N**: `[in] CBLAS_INT` number of indexed elements.
-   **alpha**: `[in] double` scalar constant.
-   **X**: `[in] double*` input array.
-   **strideX**: `[in] CBLAS_INT` index increment for `X`.
-   **offsetX**: `[in] CBLAS_INT` starting index for `X`.

```c
double stdlib_strided_dapxsumpw_ndarray( const CBLAS_INT N, const double alpha, const double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/blas/ext/base/dapxsumpw.h"
#include <stdio.h>

int main( void ) {
    // Create a strided array:
    const double x[] = { 1.0, -2.0, 3.0, -4.0, 5.0, -6.0, 7.0, -8.0 };

    // Specify the number of indexed elements:
    const int N = 8;

    // Specify a stride:
    const int strideX = 1;

>     // Fill the array:

Reviewer: ```suggestion
    // Compute the sum:
```
